### PR TITLE
Networking Fixes

### PR DESCRIPTION
**Pull request by nisdas**
- [x] Make sure an ENR has a valid tcp key saved in it before proceeding to create its multiaddress and dialing it.
- [x] Fix metadata ssz tags
- [x] Fix failing tests

### CODE REVIEW

File: beacon-chain/p2p/service.go [hunk 1]
```diff
@@ -565,6 +572,13 @@ func (s *Service) connectToBootnodes() error {
 		if err != nil {
 			return err
 		}
+		// do not dial bootnodes with their tcp ports not set
+		if err := bootNode.Record().Load(enr.WithEntry("tcp", new(enr.TCP))); err != nil {
```
Comment (shayzluf) [inline beacon-chain/p2p/service.go:576]:
```suggestion
		if err := bootNode.Record().Load(enr.WithEntry(enr.TCP.ENRKey, new(enr.TCP))); err != nil {
```

File: beacon-chain/p2p/service.go [hunk 2]
```diff
@@ -524,6 +524,13 @@ func (s *Service) processPeers(nodes []*enode.Node) []ma.Multiaddr {
 		if node.IP() == nil {
 			continue
 		}
+		// do not dial nodes with their tcp ports not set
+		if err := node.Record().Load(enr.WithEntry("tcp", new(enr.TCP))); err != nil {
```
Comment (shayzluf) [inline beacon-chain/p2p/service.go:528]:
```suggestion
		if err := node.Record().Load(enr.WithEntry(enr.TCP.ENRKey, new(enr.TCP))); err != nil {
```

File: beacon-chain/p2p/service.go (hunk 1 above)
Comment (nisdas) [reply beacon-chain/p2p/service.go:576]: `TCP` is a type not value

File: beacon-chain/p2p/service.go (hunk 2 above)
Comment (nisdas) [reply beacon-chain/p2p/service.go:528]: same here

**Comment by prestonvanloon** [comment]
We can close #5422 if this merges first.

**Pull request merged**